clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/chunk/ChunkAssembler.java

```diff
@@ -133,18 +147,37 @@ public ValueBytesAndSchemaId assembleAndGetValue(final byte[] keyBytes, final It
           continue;
         }
         // Collecting chunks
-        for (int i = 0; i < chunkKeySuffixes.length; i++) {
-          ByteBuffer byteBuffer = chunkKeySuffixes[i];
-          if (byteBuffer.equals(reusedMapperValue.chunkedKeySuffix)) {
-            byte[] valueChunk = new byte[reusedMapperValue.value.remaining()];
-            totalByteCount += valueChunk.length;
-            reusedMapperValue.value.get(valueChunk);
-            valueChunks[i] = valueChunk;
-            chunksFound++;
-            break;
+        boolean isChunkMatched = false;
+        if (isRmdChunkingEnabled && (rmdChunksFound != rmdChunkKeySuffixes.length)) {
+          for (int i = 0; i < rmdChunkKeySuffixes.length; i++) {
+            ByteBuffer byteBuffer = rmdChunkKeySuffixes[i];
+            if (byteBuffer.equals(reusedMapperValue.chunkedKeySuffix)
+                && (reusedMapperValue.replicationMetadataPayload.remaining() > 0)) {
```

Comment:
The problem here is the chunk naming formula is the same for value chunk and RMD chunk...so here we cannot distinguish between value chunk and RMD chunk actually...

Reply:
without checking the size of either value or RMD payload.

Reply:
Over slack, there is a more serious issue discovered since RMD chunks and value chunks are sharing the sam key space, and it needs to be addressed before enabling RMD chunking.